configs/configupgrade/upgrade_expr.go

```diff
@@ -168,10 +168,19 @@ Value:
 			src, moreDiags := upgradeExpr(node, filename, interp, an)
 			diags = diags.Append(moreDiags)
 			buf.Write(src)
-			if multiline {
-				buf.WriteString(",\n")
-			} else if i < len(tv.List)-1 {
-				buf.WriteString(", ")
+			if lit, ok := node.(*hcl1ast.LiteralType); ok && lit.LineComment != nil {
+				for _, comment := range lit.LineComment.List {
+					buf.WriteString(", " + comment.Text)
+				}
+				if multiline {
+					buf.WriteString("\n")
```

Comment:
I think this is missing a comma+space? :thinking:

Reply:
Oh sorry, I just re-read and I understand better on the second read. :confounded:

Reply:
I think what confused me here is that this doesn't seem to have any behavior for if `multiline` is `false`.

By how the parser works I think it's impossible to have line comments in single-line mode, so in practice that probably doesn't really matter, but perhaps it would be clearer to remove the `if multiline` condition altogether here and just always print the newline.

If a list item is a string but has no comment at all, would this logic lose its trailing comma, or am I missing something that deals with that? I think in that case `lit.LineComment.List` woudl be empty, so we'd print no commas at all.

```hcl
list = [
  "foo",
  "bar",
]
```

Reply:
I added a test to validate the behavior for a string list item with no comment, and it passed - that's captured in the `else` statement a few lines below (in this case: `else there are no comments`). I'll push that test (and removing the redundant `if multiline` - I had _thought_ it was redundant but didn't trust myself). Thanks!

Reply:
Ahh, what I missed was the `&& lit.LineComment != nil` on the first branch above. The "if (type assertion)" lead-in caused me to gloss over the rest as a "stock phrase" `if lit, ok := node.(*hcl1ast.LiteralType); ok {` and not spot that extra condition on the end.

Makes sense now! Thanks for adding the extra test.